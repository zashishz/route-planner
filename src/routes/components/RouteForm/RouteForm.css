.form-input-controls {
    display: flex;
}

.form-input-controls button {
    border: 0;
    font-weight: bold;
    background-color: #e2e2e2;
    cursor: pointer;
}