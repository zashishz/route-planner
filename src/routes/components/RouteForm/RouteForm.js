import React, { Component } from 'react';
import { maps } from '../../../shared/services/maps';

import './RouteForm.css';
class RouteForm extends Component {

    startInput = React.createRef();
    dropInput = React.createRef();

    setupAutoComplete = async () => {
        const maps = await this.props.maps();
        this.setupStartInputAutoComplete = new maps.places.Autocomplete(this.startInput);
        this.setupDropInputAutoComplete = new maps.places.Autocomplete(this.dropInput);
    }

    getRoute = () => {
        const from = this.setupStartInputAutoComplete.getPlace();
        const to = this.setupDropInputAutoComplete.getPlace();
        this.props.getRoutes(from, to);
    }

    componentDidMount() {
        this.setupAutoComplete();
    }

    render() {
        return (
            <div className="route-form">
                <div className="form-input">
                    <label>Starting Location</label>
                    <input type="text" ref={el => (this.startInput = el)} />
                </div>
                <div className="form-input">
                    <label>Drop-off point</label>
                    <input type="text" ref={el => (this.dropInput = el)} />
                </div>
                <div className="get-route-btn">
                    <button onClick={this.getRoute}>Get Route</button>
                </div>
            </div>
        )
    }
}

RouteForm.defaultProps = {
    maps
}

export default RouteForm;