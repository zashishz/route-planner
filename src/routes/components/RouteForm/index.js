import RouteForm from './RouteForm';

export { RouteForm };