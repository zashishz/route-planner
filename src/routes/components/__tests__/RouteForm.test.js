import React from "react";
import renderer from "react-test-renderer";

import { RouteForm } from "../RouteForm";

describe("Render RouteForm correctly", () => {
  it("render correctly", () => {
    const tree = renderer.create(<RouteForm />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
