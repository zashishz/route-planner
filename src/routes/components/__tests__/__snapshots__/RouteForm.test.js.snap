// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RouteForm correctly render correctly 1`] = `
<div
  className="route-form"
>
  <div
    className="form-input"
  >
    <label>
      Starting Location
    </label>
    <input
      type="text"
    />
  </div>
  <div
    className="form-input"
  >
    <label>
      Drop-off point
    </label>
    <input
      type="text"
    />
  </div>
  <div
    className="get-route-btn"
  >
    <button
      onClick={[Function]}
    >
      Get Route
    </button>
  </div>
</div>
`;
